internal/cli/atlas/quickstart/quick_start.go

```diff
@@ -130,6 +130,12 @@ func (opts *Opts) initStore(ctx context.Context) func() error {
 	}
 }
 
+func (opts *Opts) quickstartPreRun() error {
+	return opts.PreRunE(
+		opts.ValidateProjectID,
+	)
+}
+
 func (opts *Opts) PreRun(ctx context.Context, outWriter io.Writer) error {
 	opts.setTier()
```

Comment:
[q] Why didn’t you add this validation in the existing `PreRun`? we are already validating `CurrentIP` there

Reply:
The`ValidateProjectID` is the validation that throws `required flag(s) "%s" not set`, which means it's quickstart specific (other commands that use quickstart Run and PreRun manage their own flags). To make sure this validation is done only at the quickstart command itself we decouple it from `PreRun` which is publicly used by `setup`.

Reply:
ah, I see that setup is using `quickstart.PreRun` indeed. I am wondering if stop using the quickstart.PreRun would be a better way but this is no blocking

Reply:
Thnaks! The idea is to keep `PreRun` with as many validations that can be shared as possible, and keep the individual ones in a private method.